code/obj/item/gun/kinetic.dm

```diff
@@ -2209,6 +2353,68 @@ ABSTRACT_TYPE(/obj/item/survival_rifle_barrel)
 
 
 
+// assault
+/obj/item/gun/kinetic/m16
+	name = "\improper M16"
+	desc = "An iconic weapon design that doesn't need explanation. Besides this one... Uh. Uses 5.56 rounds."
+	icon = 'icons/obj/items/guns/kinetic48x32.dmi'
+	icon_state = "m16"
+	item_state = "m16"
+	wear_image_icon = 'icons/mob/clothing/back.dmi'
+	flags =  FPRINT | TABLEPASS | CONDUCT | USEDELAY
+	c_flags = EQUIPPED_WHILE_HELD | ONBACK
+	force = MELEE_DMG_RIFLE
+	contraband = 8
+	ammo_cats = list(AMMO_AUTO_556)
+	max_ammo_capacity = 15
+	auto_eject = TRUE
+	two_handed = TRUE
+	can_dual_wield = FALSE
+	spread_angle = 0
+	fire_animation = TRUE
+	has_empty_state = TRUE
+	default_magazine = /obj/item/ammo/bullets/assault_rifle
+
+	New()
+		ammo = new default_magazine
+		set_current_projectile(new/datum/projectile/bullet/assault_rifle)
+		projectiles = list(current_projectile,new/datum/projectile/bullet/assault_rifle/burst)
+		..()
+
+	attackby(obj/item/ammo/bullets/b, mob/user)  // has to account for whether regular or armor-piercing ammo is loaded AND which firing mode it's using
+		var/obj/previous_ammo = ammo
+		var/mode_was_burst = (istype(current_projectile, /datum/projectile/bullet/assault_rifle/burst))  // was previous mode burst fire?
+		..()
+		if(previous_ammo.type != ammo.type)  // we switched ammo types
```

Comment:
this whole bit of checking ammo sucks ass. probably better with me attacking it by finalizing my kinetics branch